### PR TITLE
Modals & Drawers & Oh So Much More

### PR DESCRIPTION
**Pull request by piazzatron**
### Description
- Adds modals + drawers
- Adds a new twitter embed library for embedding a tweet in a modal
- Hooks things up to optimizely
- Various UI tweaks
- IG + twitter verif buttons probably don't work yet! (that's next up)
- Changes to @sddioulde's icon popover
- This is huge and I'm sorry

### Dragons

None

### How Has This Been Tested?

Locally

### CODE REVIEW

File: src/containers/audio-rewards-page/components/modals/ModalDrawer.module.css [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+  fill: var(--static-white);
+}
+
+.drawer {
+  padding: 32px 12px;
+  width: 100%;
+}
+
+.drawerTitle {
+  font-weight: 800;
```
Comment (jowlee) [inline src/containers/audio-rewards-page/components/modals/ModalDrawer.module.css:25]: we don't have 800

File: src/containers/audio-rewards-page/components/ButtonWithArrow.module.css [hunk 2]
```diff
@@ -0,0 +1,7 @@
+.rewardButton {
+  padding: 8px 12px;
+  height: 34px !important;
+}
+.buttonIcon {
+  margin: 0 2px 0 0px !important;
```
Comment (jowlee) [inline src/containers/audio-rewards-page/components/ButtonWithArrow.module.css:6]: Would you still need `!important` if this was `.rewardButton .buttonIcon {` 
I'm not sure, but I don't like important, but also I use it it a lot 🙃 . 
Feel free to leave as it.

File: src/containers/audio-rewards-page/components/modals/TrendingRewards.tsx [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+            }}
+          />
+        </div>
+      </div>
+      <ButtonWithArrow
+        text={textMap[modalType].button}
+        onClick={onButtonClick}
+        className={styles.button}
+      />
+      <a href='' className={styles.terms}>
```
Comment (jowlee) [inline src/containers/audio-rewards-page/components/modals/TrendingRewards.tsx:147]: Is this supposed to be a link?

File: src/containers/audio-rewards-page/components/modals/VerifiedUpload.tsx [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+      <span className={styles.title}>{messages.step1Title}</span>
+      <span className={styles.subtitle}>{messages.step1Subtitle}</span>
+      <div className={styles.verifyButtons}>
+        <TwitterButton
+          textLabel={messages.verifyTwitterButton}
+          size='large'
+          className={styles.twitterButton}
+        />
+        <InstagramButton
+          // TODO: handle these
```
Comment (jowlee) [inline src/containers/audio-rewards-page/components/modals/VerifiedUpload.tsx:68]: Is this intentionally left not finished to be done later?

**Review by jowlee — APPROVED** [review]
wow, so much. code looks good. Is this running somewhere where I can play around w/ it?

File: src/containers/audio-rewards-page/components/modals/ModalDrawer.module.css (hunk 1 above)
Comment (piazzatron) [reply src/containers/audio-rewards-page/components/modals/ModalDrawer.module.css:25]: will fix

File: src/containers/audio-rewards-page/components/ButtonWithArrow.module.css (hunk 2 above)
Comment (piazzatron) [reply src/containers/audio-rewards-page/components/ButtonWithArrow.module.css:6]: will test

File: src/containers/audio-rewards-page/components/modals/TrendingRewards.tsx (hunk 3 above)
Comment (piazzatron) [reply src/containers/audio-rewards-page/components/modals/TrendingRewards.tsx:147]: yeah broken for now, gonna add on next iteration

File: src/containers/audio-rewards-page/components/modals/VerifiedUpload.tsx (hunk 4 above)
Comment (piazzatron) [reply src/containers/audio-rewards-page/components/modals/VerifiedUpload.tsx:68]: yep. sry was unclear

File: src/containers/audio-rewards-page/components/ButtonWithArrow.module.css (hunk 2 above)
Comment (piazzatron) [reply src/containers/audio-rewards-page/components/ButtonWithArrow.module.css:6]: no dice :(

**Pull request merged**